src/Compilers/CSharp/Portable/Symbols/AbstractTypeMap.cs

```diff
@@ -232,6 +235,31 @@ private PointerTypeSymbol SubstitutePointerType(PointerTypeSymbol t)
             return new PointerTypeSymbol(pointedAtType);
         }
 
+        private FunctionPointerTypeSymbol SubstituteFunctionPointerType(FunctionPointerTypeSymbol f)
+        {
+            bool paramTypesChanged = false;
+            var substitutedParamTypes = ArrayBuilder<TypeWithAnnotations>.GetInstance(f.Signature.ParameterCount);
+            foreach (var param in f.Signature.Parameters)
+            {
+                var substitutedType = param.TypeWithAnnotations.SubstituteType(this);
+                if (!param.TypeWithAnnotations.IsSameAs(substitutedType))
+                {
+                    paramTypesChanged = true;
+                }
+                substitutedParamTypes.Add(substitutedType);
+            }
+
+            var substitutedReturnType = f.Signature.ReturnTypeWithAnnotations.SubstituteType(this);
+
+            if (paramTypesChanged || !f.Signature.ReturnTypeWithAnnotations.IsSameAs(substitutedReturnType))
+            {
+                f = f.SubstituteTypeSymbol(substitutedReturnType, substitutedParamTypes);
```

Comment:
>substitutedParamTypes [](start = 66, length = 21)

It is probably better to pass an immutable array as we do for substituted generic type arguments